suuchi-core/src/main/scala/in/ashwanthkumar/suuchi/store/VersionedStore.scala

```diff
@@ -107,4 +111,8 @@ class VersionedStore(store: Store, versionedBy: VersionedBy, numVersions: Int, c
 
   private def removeData(key: Array[Byte], version: Long) = store.remove(dkey(key, version))
   private def removeVersion(key: Array[Byte]) = store.remove(vkey(key))
+
```

Comment:
Selva - we talked about adding a new method `scanVersions` (with 2 overloaded variants like `scan`) that does prefix based on `VersionedStore.vkey` in VersionedStore. I find that missing. Can you please add that too? 

We don't have an immediate need for it but that goes with symmetry for VersionedStore given we can scan both just for versions and for records.